cdap-master/src/test/java/co/cask/cdap/data/tools/UpgradeToolTest.java

```diff
@@ -24,7 +25,10 @@
 public class UpgradeToolTest {
   @Test
   public void testInjector() throws Exception {
-    new UpgradeTool();
+    Injector upgradeToolInjector = new UpgradeTool().createInjector();
+    // should not throw exception
+    // Test the UpgradeDatasetServiceManager injector creation
+    upgradeToolInjector.getInstance(UpgradeDatasetServiceManager.class);
```

Comment:
hmmm, I don't think this will test the injector in ``UpgradeToolDatasetManager``. You will have to call ``UpgradeToolDatasetManager.createInjector()`` to test that, no?